File: Chauffeur/Deliverables/ChangeAliasDeliverable.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Umbraco.Core.Services;

namespace Chauffeur.Deliverables
{
    [DeliverableName("change-alias")]
    [DeliverableAlias("ca")]
    public sealed class ChangeAliasDeliverable : Deliverable
    {
        private readonly IContentTypeService contentTypeService;

        public ChangeAliasDeliverable(
            TextWriter writer,
            TextReader reader,
            IContentTypeService contentTypeService
            )
            : base(reader, writer)
        {
            this.contentTypeService = contentTypeService;
        }

        public override async Task<DeliverableResponse> Run(string command, string[] args)
        {
            if (args.Length != 3)
            {
                await Out.WriteLineAsync("Invalid arguments, expected format of `change-alias <what> <old> <new>`");
                return DeliverableResponse.Continue;
            }

            var what = args[0].ToLower();

            var old = args[1];
            var @new = args[2];

            switch (what)
            {
                case "document-type":
                case "doc-type":
                case "dt":
                    var ct = contentTypeService.GetContentType(old);
                    ct.Alias = @new;
                    contentTypeService.Save(ct);
                    break;

                default:
                    await Out.WriteLineFormattedAsync("Presently we cannot change the alias of `{0}`. Run `help ca` to see what can be changed", what);
                    break;
            }

            return DeliverableResponse.Continue;
        }
    }
}

File: src/Chauffeur/Deliverables/ChangeAliasDeliverable.fs
namespace Chauffeur.Deliverables

open Chauffeur
open FSharp.Control.Tasks.V2
open Umbraco.Core.Services

[<DeliverableName("change-alias")>]
[<DeliverableAlias("ca")>]
type ChangeAliasDeliverable(reader, writer, contentTypeService : IContentTypeService) =
    inherit Deliverable(reader, writer)

    override __.Run _ args =
        match args with
        | [|"document-type"; old; ``new``|]
        | [|"doc-type"; old; ``new``|]
        | [|"dt"; old; ``new``|] -> task {
            let ct = contentTypeService.Get old
            ct.Alias <- ``new``
            contentTypeService.Save ct
            return DeliverableResponse.Continue }

        | _ -> task {
            do! writer.WriteLineAsync "In valid arguments, expected format of `change-alias what old new`."
            return DeliverableResponse.Continue }